src/models/jwtPermissions/index.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+const { JWTVerifier } = require('@storyous/storyous-jwt');
+const fs = require('fs');
+const path = require('path');
+
+// eslint-disable-next-line import/no-unresolved
+const { permissionHelper, fetch } = require('../../index');
+const { NotSufficientPermissions } = require('./customErrors');
+const { isProduction, isTesting } = require('../../config');
+
+const publicKeys = {};
+let _publicKeyUrl;
+let keyName;
+if (isProduction()) { _publicKeyUrl = 'http://publickey.storyous.com'; keyName = 'publicProduction'; } else
+if (isTesting()) {
+    _publicKeyUrl = 'http://127.0.0.1:3010/getPublicKey'; keyName = 'publicTesting';
+} else { _publicKeyUrl = 'http://publickey.storyous.com'; keyName = 'publicTest'; }
+const publicKeyPath = path.join(__dirname, '..', '..', '..', 'publicKeys', keyName);
+
+/**
+ *
+ * @param {null|string}authorization
+ * @returns {string}
+ */
+const parseAuthorization = (authorization) => {
+    if (!authorization) { return ''; }
+    const parsedAuthorization = authorization.split(' ').filter((i) => i !== ' ');
+    return parsedAuthorization[1];
+};
+
+/**
+ *
+ * @param {object}payload
+ * @returns {{decodedPayload: {merchantId:string, decodedPermissions: (boolean[])}, originalPayload}}
+ */
+function decodePayload (payload) {
+    const { permissions, merchantId } = payload;
+    const decodedPermissions = permissions ? permissionHelper.decodeData(permissions) : [];
+    return { originalPayload: payload, decodedPayload: { merchantId, decodedPermissions } };
+
+}
+
+/**
+ *
+ * @param {string|null}publicKeyUrl
+ * @returns {Promise<string>}
+ */
+async function getJwt (publicKeyUrl = null) {
+    if (publicKeys[publicKeyUrl]) { return publicKeys[publicKeyUrl]; }
+    const publicKeyLocal = new Promise((resolve) => {
+        setTimeout(resolve, 5000, fs.readFileSync(publicKeyPath).toString());
+    });
+    const publicKeyLoaded = fetch.text(publicKeyUrl);
+
+    publicKeys[publicKeyUrl] = await Promise.race([publicKeyLocal, publicKeyLoaded]);
+    return publicKeys[publicKeyUrl];
+}
+
+/**
+ *
+ * @param {string|null}publicKeyUrl
+ * @returns {(function(*, *): Promise<void>)|*}
+ */
+exports.validateJwtWithPermissions = ({ publicKeyUrl = null }) => async (ctx, next) => {
+    if (!publicKeyUrl) publicKeyUrl = _publicKeyUrl;
```

Comment:
Remove this line and use it default value as

`exports.validateJwtWithPermissions = ({ publicKeyUrl = _publicKeyUrl }) => async (ctx, next) => {`

Reply:
+ same on line 48 (or do not allow default value at all if you want to call getJwt always with publicKeyUrl)

Reply:
done